static/js/publisher/release/releasesController.js

```diff
@@ -26,10 +30,57 @@ export default class ReleasesController extends Component {
       //  }
       // }
       pendingReleases: {},
-      pendingCloses: []
+      pendingCloses: [],
+      // list of selected revisions, to know which ones to render selected
+      selectedRevisions: []
     };
   }
 
+  // update list of architectures based on revisions released (or selected)
+  getArchsFromReleasedChannels(releasedChannels) {
+    let archs = [];
+    Object.keys(releasedChannels).forEach(channel => {
+      Object.keys(releasedChannels[channel]).forEach(arch => {
+        archs.push(arch);
+      });
+    });
+
+    // make archs unique and sorted
+    archs = archs.filter((item, i, ar) => ar.indexOf(item) === i);
+
+    return archs.sort();
+  }
+
+  selectRevision(revision) {
+    this.setState((state) => {
+      const releasedChannels = state.releasedChannels;
+      // TODO: export as a constant?
+      const UNASSIGNED = 'unassigned';
```

Comment:
I've been thinking we should have a "constants" file somewhere to keep all the things like this together. Might be useful when we start reduxing.